kafka/integration_test.go

```diff
@@ -1659,7 +1659,131 @@ func (its *IntegrationTestSuite) TestAdminACLs() {
+	t := its.T()
+	rand.Seed(time.Now().Unix())
+
+	conf := &ConfigMap{"bootstrap.servers": testconf.Brokers}
+	if err := conf.updateFromTestconf(); err != nil {
+		t.Fatalf("Failed to update test configuration: %s\n", err)
+	}
+
+	admin, err := NewAdminClient(conf)
+	if err != nil {
+		t.Fatalf("Failed to create Admin client: %s\n", err)
+	}
+	defer admin.Close()
+
+	// Create some topics.
+	numTopics := 3
+	topics := make([]string, 0)
+	topicSpec := make([]TopicSpecification, 0)
+
+	for i := 0; i < numTopics; i++ {
+		topic := fmt.Sprintf("%s-%d", testconf.Topic, rand.Intn(100000))
+		topics = append(topics, topic)
+		topicSpec = append(
+			topicSpec, TopicSpecification{Topic: topic, NumPartitions: i + 1})
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+	topicResult, err := admin.CreateTopics(ctx, topicSpec)
+
+	if err != nil {
+		t.Fatalf("Failed to create topics: %s\n", err)
+	}
+
+	for i := 0; i < numTopics; i++ {
+		if topicResult[i].Error.Code() != ErrNoError {
+			t.Fatalf("Failed to create topic %s: %s", topicSpec[i].Topic, topicResult[i].Error)
+		}
+	}
+
+	// Join a consumer group and subscribe to the created topics,
+	// commit some offsets to read later.
+	group := fmt.Sprintf("%s-%d", testconf.GroupID, rand.Intn(100000))
+	conf = &ConfigMap{
+		"bootstrap.servers": testconf.Brokers,
+		"group.id":          group,
+		"auto.offset.reset": "end",
+	}
+	conf.updateFromTestconf()
+
+	consumer, err := NewConsumer(conf)
+	if err != nil {
+		t.Fatalf("Failed to create consumer: %s\n", err)
+	}
+	defer consumer.Close()
+
+	if err = consumer.SubscribeTopics(topics, nil); err != nil {
+		t.Fatalf("Failed to subscribe to topics: %s\n", err)
+	}
+
+	// Poll for a while, wait for rebalance.
+	consumer.Poll(10000)
+
+	for i := 0; i < numTopics; i++ {
+		for j := 0; j < topicSpec[i].NumPartitions; j++ {
+			_, err = consumer.CommitOffsets([]TopicPartition{
+				{
+					Topic:     &topics[i],
+					Partition: int32(j),
+					Offset:    Offset((i * topicSpec[i].NumPartitions) + j),
+				},
+			})
+			if err != nil {
+				t.Fatalf("Could not commit for %s:%d: %s\n", topics[i], j, err)
+			}
+		}
+	}
+
+	// List consumer group offsets.
+	ctx, cancel = context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+	offsets, err := admin.ListConsumerGroupOffsets(
+		ctx,
+		[]ConsumerGroupTopicPartitions{{Group: group}},
+		SetAdminRequireStableOffsets(true))
+
+	if err != nil {
+		t.Fatalf("Failed to get consumer group offsets: %s\n", err)
+	}
+
+	if len(offsets.ConsumerGroupsTopicPartitions) != 1 {
+		t.Fatal("Consumer group offsets is empty")
+	}
+
+	toppars := offsets.ConsumerGroupsTopicPartitions[0].Partitions
+
+	// Use linear search - okay for small numTopics. Since the returned list
+	// is not ordered in any particular way.
+	matchedToppars := 0
+	for i := 0; i < numTopics; i++ {
+		for k := 0; k < len(toppars); k++ {
```

Comment:
Just making sure that `len(toppars)` will return the cached length right? Its not like C which calculates the length in every loop.

Reply:
Yes, the length is stored in a precomputed manner, so no issues like that for go